tests/sentry/test_event_manager.py

```diff
@@ -125,20 +125,6 @@ def test_saves_event_mapping_when_sampled(self, should_sample):
             event_id=event_id,
         ).exists()
 
-    @mock.patch('sentry.event_manager.should_sample')
-    def test_sample_feature_flag(self, should_sample):
-        should_sample.return_value = True
-
-        manager = EventManager(self.make_event())
-        with self.feature('projects:sample-events'):
-            event = manager.save(1)
-        assert event.id
-
-        manager = EventManager(self.make_event())
-        with self.feature({'projects:sample-events': False}):
-            event = manager.save(1)
-        assert not event.id
```

Comment:
I'm pretty sure this test was always a false positive — it was being caught as a duplicate.

I'm also not sure why *disabling* sampling would have caused the `assert not event.id`, so this test just doesn't really make sense to me in general. Maybe I'm missing something?

Reply:
if should_sample returned True that means "sample this event" and the Event would be returned but it'd have no ID as it wasnt persisted in the database

Reply:
I don't really understand how that relates to this specific test (as implemented, at least.)

In this case, with the switch on, an ID was returned (contrary to what you just wrote above) because the result of `should_sample` is overridden here since the created group is new (otherwise it'd have the same effect as the unrelated issue this patch is fixing, where the group is created with no events):

https://github.com/getsentry/sentry/blob/72d040cc34d10bfd2f95df42852aa3d14da0a80a/src/sentry/event_manager.py#L1080-L1081

Then, with the switch off, it the event doesn't have an ID (again contrary to the above) because it has a duplicate event ID (UUID, not primary key) and no event is created.
 (Note: This patch does change *that* behavior, since `save` now returns the event that was retrieved from the database, similar to `get_or_create`, instead of `save` returning the event that *wasn't* saved.)